dvci/iis-shc-issuer
# Dvci 150 Dockerize IIS Issuer

**Summary:**
Dockerizes IIS-SHC-Issuer, including containers for the [IIS-Sandbox](https://github.com/dvci/IIS-Sandbox) and the [v2-to-fhir](https://github.com/dvci/hl7v2-fhir-vaccine-credential-converter) converter.

**To test:**
First, run:
`echo IIS-Sandbox.git hl7v2-fhir-vaccine-credential-converter.git | xargs -n1 | xargs -I{} git clone --recursive https://github.com/dvci/{}` to clone the submodules needed for the two dependent services.

Then, run: `docker-compose up -d` to build the Docker images and run the containers.

Visit  http://localhost:5000 to see our working front-end IIS issuer app.

To check dependant services, visit http://localhost:3000/api-docs/ to see that the v2-to-fhir api is up and running. Visit http://localhost:8081/iis-sandbox/ to see that our dockerized version of the iis-sandbox is live.

~_Note:_ 
Creating the docker containers required changing the host environment variables to local servers instead of our internal MITRE ones. This required changing how our tests are mocked to allow for different hosts, causing changes to all of the iis_shc_issuer test files and the vcr_cassettes files.~
This was fixed in a previous PR.
